test/integration/scripting_spec.mjs

```diff
@@ -2216,20 +2207,12 @@ describe("Interaction", () => {
           await page.click(getSelector("55R"));
           await page.type(getSelector("55R"), "Hello", { delay: 10 });
           await page.click(getSelector("56R"));
-          // eslint-disable-next-line no-restricted-syntax
-          await waitForTimeout(10);
 
           await page.click(getSelector("55R"));
           await page.type(getSelector("55R"), " World", { delay: 10 });
-          // eslint-disable-next-line no-restricted-syntax
-          await waitForTimeout(10);
 
           await otherPages[i].bringToFront();
-          // eslint-disable-next-line no-restricted-syntax
-          await waitForTimeout(100);
```

Comment:
I'm not sure it's a good idea to remove them here.
Even if the test is ok without, the goal was to be sure that nothing is executed when getting focus and it's pretty hard to wait on "nothing happened", the only way I can imagine is to wait for a timeout.

Reply:
Isn't activating the tab already enough to ensure that anything that would happen is already executed by the time `bringToFront` is done? However, if you're not sure about this I'm also OK with keeping this occurrence for now (we have a similar situation of "waiting for nothing to happen" elsewhere and I'm also not sure how to reliably handle that one either).

Reply:
Maybe this one could be removed: I don't know, but the second should be kept.

Reply:
I just thought of a possible solution for such cases (but that's for a follow-up, so I'll put back these timeouts for now as you suggested): perhaps we could emit an event for the tests only when the page got activated/focused and all listeners are done. That way in the tests we can wait for this event to be emitted and then we'll be sure that any handlers have run. We'd have to check how feasible that is to implement, but if possible it would give us something to await here without having to wait for a timeout.

Reply:
I know timeouts aren't ideal but I don't think doing such a thing is better.
We could trigger an event when the tab is brought back but maybe the focus handler will be triggered after and then we won't catch a regression.
We must have a way to be 100% sure that whatever stuff hasn't been called !!
In the current state of my mind, the only way to solve the "wait on nothing" pattern is to wait for a timeout.
That said if they're rare enough we can maybe increase them to 500ms...

Reply:
I have put them back for this test in the new commit. For now that indeed seems better, and later we can re-evaluate if we can have a better way (somehow awaiting all focus handlers would be possible) or just decide to keep them (that'd also probably be fine for the difficult cases such as this test, and as long as we get rid of all other occurrences where it's easier then it's already quite an improvement).